File: Buttons/app/src/main/java/demoapp/codingblocks/com/buttons/MainActivity.java
package demoapp.codingblocks.com.buttons;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener{
    Button bt1,bt2,bt3,bt4;
    int a=10,b=10;
    TextView tv;
    EditText t1,t2;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bt1=findViewById(R.id.addbutton); //Defining button1 otherwise it will get garbage value
        bt2=findViewById(R.id.subbutton);
        bt3=findViewById(R.id.mulbutton);
        bt4=findViewById(R.id.divbutton);
        tv=findViewById(R.id.tex);
        t1=findViewById(R.id.edit1);
        t2=findViewById(R.id.edit2);
//        bt1.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                Toast.makeText(MainActivity.this,"Button 1 click!!",Toast.LENGTH_LONG).show();
//                //We have to mention MainActivity otherwise it will take reference from View class instead of MainActivity class
//            }
//        });
        bt1.setOnClickListener(this);
        bt2.setOnClickListener(this);
        bt3.setOnClickListener(this);
        bt4.setOnClickListener(this);

    }

    @Override
    public void onClick(View v) {

        if(!t1.getText().toString().equals("")&&!t2.getText().toString().equals("")){
            a=Integer.parseInt(t1.getText().toString());
            b=Integer.parseInt(t2.getText().toString());
        }
        else{
            Toast.makeText(this,"Number must not be empty",Toast.LENGTH_SHORT).show();
        }
        if(v.getId()==R.id.addbutton) {
            tv.setText(a + b + "");
            Toast.makeText(this, "Button1click", Toast.LENGTH_SHORT).show();
        }
        if(v.getId()==R.id.subbutton) {
            tv.setText(a - b + "");
            Toast.makeText(this, "Button2click", Toast.LENGTH_SHORT).show();
        }
        if(v.getId()==R.id.mulbutton) {
            tv.setText(a * b + "");
            Toast.makeText(this, "Button3click", Toast.LENGTH_SHORT).show();
        }
        if(v.getId()==R.id.divbutton) {
            tv.setText(a / b + "");
            Toast.makeText(this, "Button4click", Toast.LENGTH_SHORT).show();
        }
    }

//    public void button2Click(View view) {
//        Toast.makeText(this,"Button2click!!",Toast.LENGTH_SHORT).show();
//
//    }
//
//    public void button3Click(View view) {
//        Toast.makeText(this,"Button3click!!",Toast.LENGTH_SHORT).show();
//        view.setBackgroundResource(R.color.green);
//    }
//
//    public void button4click(View view) {
//        Toast.makeText(this,"Button4click!!",Toast.LENGTH_SHORT).show();
//        view.setBackgroundResource(R.color.cyan);
//    }

}

File: Stopwatch/app/src/main/java/codingblocks/com/stopwatch/MainActivity.kt
package codingblocks.com.stopwatch

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        val uiScope = CoroutineScope(Dispatchers.Main)

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var stop=0
        var msec=0
        var sec=0
        bt1.setOnClickListener {
            stop=0

            uiScope.launch {
                for (j in sec..59) {
                    for (i in msec..99) {
                        if(stop==1) {
                            break
                        }
                        delay(8.toLong())
                        if(i<10)
                            tv3.text="0"+"$i"
                        else
                            tv3.text="$i"
                    }
                    if(stop==1) {
                        break
                    }
                    if(j<10)
                        tv2.text = "0"+(j.toString().toInt()+1).toString()+"."
                    else
                        tv2.text = (j.toString().toInt()+1).toString()+"."
                    msec=0
                }
            }
        }
        bt2.setOnClickListener {
            stop=1
            bt1.text="resume"
            sec=tv2.text.substring(0,tv2.text.length-1).toInt()
            msec=tv3.text.toString().toInt()
        }
        bt3.setOnClickListener {
            stop=1
            bt1.text="Start"
            tv2.text="00."
            tv3.text="00"
            msec=0
            sec=0
        }
    }
    }
